packages/flutter_tools/lib/src/commands/build_web.dart

```diff
@@ -126,14 +127,22 @@ class BuildWebCommand extends BuildSubCommand {
   @override
   final String description = 'Build a web application bundle.';
 
+  bool get _compileWasm => boolArg('wasm');
+
+  @override
+  Future<CustomDimensions> get usageValues async {
+    final String compiler = _compileWasm ? 'dart2wasm' : 'dart2js';
+    return CustomDimensions(buildEventSettings: 'compiler:$compiler');
```

Comment:
Sorry can't dig through the code right now, but wondering if setting this to `compiler:SOMETHING` is the same thing we do elsewhere with this dimension?

Reply:
We don't use `buildEventSettings` ANYWHERE else. Trying to future-proof a bit.

I guess we can invent something when we need to. Happy to remove if you think we should, @zanderso

Reply:
Took a look. `buildEventSettings` is normally only set for `BuildEvent`s (from https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/reporting/events.dart#L152), and not attached to analytics commands/screens. Let me take a look to see if I can make a better suggestion.

Reply:
So, the thing about attaching the compiler info to the command/screen rather than to an event is that you will miss invocations of the compiler that aren't generated starting from the `build web` command, for example from `run` or `assemble`. Those might not be a concern for web now, but maybe they will be in the future.

To future proof what it looks like you're trying to do, I'd suggest reverting the change to this file, and instead creating a `BuildEvent` adjacent to the existing `sendTiming` you added earlier, and taking a look at the fields of the `BuildEvent` to see if anything there is appropriate for the info you want to capture.